addons/editor-number-arrow-keys/userscript.js

```diff
@@ -44,46 +44,33 @@ export default async function ({ addon }) {
         numStr = numArrFiltered.join("");
       }
     }
-    return Number(numStr) * (isNumberNegative ? -1 : 1);
-  };
-
-  const normalizeNumber = (numStr) => {
-    const isNumberNegative = numStr[0] === "-";
-    const numStrPositive = isNumberNegative ? numStr.substring(1) : numStr;
-
-    let normalizedNum = numStrPositive;
 
     // Adds zero before the decimal point if necessary (.1 → 0.1)
-    if (normalizedNum[0] === ".") {
-      normalizedNum = "0" + normalizedNum;
-    }
-
-    // Removes leading zeros (02.25 → 2.25)
-    if (/^0*$/.test(numStrPositive.split(".")[0])) {
-      // Case where integerPart = (0 or 00 or 000, etc...)
-      const decimalPart = numStrPositive.split(".")[1] || "";
-      normalizedNum = `0.${decimalPart}`;
-    } else {
-      normalizedNum = normalizedNum.replace(/^0*|0*$/, "");
+    if (numStr[0] === ".") {
+      numStr = "0" + numStr;
     }
 
     // Removes trailing zeros (2.250 → 2.25)
-    if (numStrPositive.includes(".")) {
-      normalizedNum = normalizedNum.replace(/0*$/, "");
+    if (numStr.includes(".")) {
+      numStr = numStr.replace(/0*$/, "");
     }
 
     // Removes the decimal point if it's the last character (2. → 2)
-    if (normalizedNum.endsWith(".")) {
-      normalizedNum = normalizedNum.slice(0, -1);
+    if (numStr.endsWith(".")) {
+      numStr = numStr.slice(0, -1);
     }
 
-    return (isNumberNegative ? "-" : "") + normalizedNum;
+    return numStr ? (isNumberNegative ? "-" : "") + numStr : 0;
   };
 
   const isValidNumber = (numStr) => {
-    if (numStr.length > 10) return false;
-    if (amountOfDecimals(numStr) > 5) return false;
-    return normalizeNumber(numStr) === Number(numStr).toString();
+    if (numStr.length > 100) return false;
```

Comment:
You know, are 100-digit numbers really fine-tuned every day? Not sure if the limit needs to be _this_ high. And such numbers will still lose precision when used in the project anyway. Maybe somewhere closer to 30 would be sensible?